pkg/adapter/anchore/client/client.go

```diff
@@ -38,10 +43,85 @@ type ClientConfig struct {
 }
 
 func getNewRequest(clientConfiguration *ClientConfig) *gorequest.SuperAgent {
+	reloadCredential(clientConfiguration)
 	timeout := time.Duration(clientConfiguration.TimeoutSeconds) * time.Second
 	return gorequest.New().TLSClientConfig(&tls.Config{InsecureSkipVerify: clientConfiguration.TLSVerify}).SetBasicAuth(clientConfiguration.Username, clientConfiguration.Password).Timeout(timeout)
 }
 
+// Reload the anchore user name and password if AWS Secret Manager is used
+func reloadCredential(clientConfiguration *ClientConfig) {
```

Comment:
Rather than a reload, I think a credential factory model makes more sense. It would take a clientConfiguration and return a set of credentials. The default factory would return the credential in the config, but the AWS Credential Factory would return the credentials from the SecretManager. I think that would make testing the individual parts a little easier and allow a clean extension point later to use other credential managers or service accounts.

Reply:
@zhill I am totally fine with a factory pattern, can you draft the codes structure and I will push an aws implementation.